src/rp2_common/hardware_irq/irq_handler_chain.S

```diff
@@ -54,7 +54,7 @@ irq_handler_chain_slots:
 .endr
 
 irq_handler_chain_first_slot:
-    push {lr}         // Save EXC_RETURN token, so `pop {pc}` will return from interrupt
```

Comment:
I agree this is the best fix, however it is a little obscure. Could you drop a comment explaining that this is required to maintain stack alignment after the hardware has aligned the stack when writing the exception frame?